elements/elix-mixins/src/symbols.js

```diff
@@ -0,0 +1,113 @@
+import createSymbol from './createSymbol';
+
+/**
+ * A collection of Symbol objects for standard component properties and methods.
+ *
+ * These Symbol objects are used to allow mixins and a component to internally
+ * communicate, without exposing these properties and methods in the component's
+ * public API.
+ *
+ * To use these Symbol objects in your own component, include this module and
+ * then create a property or method whose key is the desired Symbol.
+ *
+ *     import 'SingleSelectionMixin' from 'elix-mixins/src/SingleSelectionMixin';
+ *     import 'symbols' from 'elix-mixins/src/symbols';
+ *
+ *     class MyElement extends SingleSelectionMixin(HTMLElement) {
+ *       [symbols.itemSelected](item, selected) {
+ *         // This will be invoked whenever an item is selected/deselected.
+ *       }
+ *     }
+ *
+ * @module symbols
+ */
+const symbols = {
+
+  /**
+   * Symbol for the `defaults` property.
+   *
+   * This property can be used to set or override defaults that will be applied
+   * to a new component instance. When implementing this property, take care to
+   * first acquire any defaults defined by the superclass. The standard idiom is
+   * as follows:
+   *
+   *     get [symbols.defaults]() {
+   *       const defaults = super[symbols.defaults] || {};
+   *       // Set or override default values here
+   *       defaults.customProperty = false;
+   *       return defaults;
+   *     }
+   *
+   * @var {object} defaults
+   */
+  defaults: createSymbol('defaults'),
+
+  /**
+   * Symbol for the `raiseChangeEvents` property.
+   *
+   * This property is used by mixins to determine whether they should raise
+   * property change events. The standard HTML pattern is to only raise such
+   * events in response to direct user interactions. This property can be used
+   * to manage events as follows.
+   *
+   * First, UI event listeners should set this property to `true` at the start
+   * of the event handler, then `false` at the end:
+   *
+   *     this.addEventListener('click', event => {
+   *       this[symbols.raiseChangeEvents] = true;
```

Comment:
In our discussion in today's core team meeting, there was general support for keeping the project's code as self-evident as possible.

At the same time, we want to make sure we're documenting the story behind use of a flag like `symbols.raiseChangeEvents`. Per a suggestion from @chiefcll and @JohnRiv, I've added documentation to the jsDoc comments for that symbol explaining its need and use. The documentation references the relevant [Property Change Events](https://github.com/webcomponents/gold-standard/wiki/Property%20Change%20Events) item on the Gold Standard checklist.

As we build our project documentation, we'll make sure that background information like that is readily available, so that new people coming to the code will not only understand what the code is doing, but why it is doing it.